Sources/ProcedureKit/Group.swift

```diff
@@ -37,20 +37,53 @@ open class GroupProcedure: Procedure, ProcedureQueueDelegate {
 
     internal let queue = ProcedureQueue()
 
-    fileprivate let finishing = BlockOperation { }
-
-    fileprivate var groupErrors = Protector(GroupErrors())
-    fileprivate var groupChildren: Protector<[Operation]>
-    fileprivate var groupIsFinishing = false
-    fileprivate var groupFinishLock = NSRecursiveLock()
-    fileprivate var groupIsSuspended = false
-    fileprivate var groupSuspendLock = NSLock()
-    fileprivate var groupIsAddingOperations = DispatchGroup()
+    fileprivate let initialChildren: [Operation]
     fileprivate var groupCanFinish: CanFinishGroup!
+    fileprivate var groupStateLock = PThreadMutex()
+
+    // Protected private properties
+    fileprivate var _groupErrors = GroupErrors()
+    fileprivate var _groupChildren: [Operation]
+    fileprivate var _groupIsFinishing = false
+    fileprivate var _groupIsSuspended = false
 
     /// - returns: the operations which have been added to the queue
     final public var children: [Operation] {
-        get { return groupChildren.read { $0 } }
+        get { return groupStateLock.withCriticalScope { _groupChildren } }
+    }
+
+    /**
+     The default service level to apply to the GroupProcedure and its child operations.
+     
+     This property specifies the service level applied to the GroupProcedure itself, and to
+     operation objects added to the GroupProcedure.
+     
+     If the added operation object has an explicit service level set, that value is used instead.
+     
+     For more, see the NSOperation and NSOperationQueue documentation for `qualityOfService`.
+     */
+    @available(OSX 10.10, iOS 8.0, tvOS 8.0, watchOS 2.0, *)
```

Comment:
Good catch - that code was previously elsewhere in the file (was moved because it's `open` to resolve a visibility scope warning), but this should indeed be fixed.